prow/jenkins/jenkins.go

```diff
@@ -388,6 +388,31 @@ func (c *Client) doRequest(method, path string) (*http.Response, error) {
 	return c.client.Do(req)
 }
 
+// getJobName generates the correct job name for this job type
+func getJobName(spec *prowapi.ProwJobSpec) string {
+	var jobName string
+
+	if spec.JenkinsSpec != nil && spec.JenkinsSpec.GithubBasedProject {
```

Comment:
This also assumes that `Refs` is non-nil. That should be true since this is only configurable for presubmits and postsubmits currently, but it would be good to make that assumption explicit so that this doesn't panic if a malformed PJ is encountered.